### ISSUE TITLE
Field validation happening immediately on focus?

### ISSUE THREAD

**Issue by thisisjamessmith** [verbb/formie#2066]
### Describe the bug

I'm finding that field validation is happening as soon as I focus a field. Is this expected behaviour? From a UX perspective, I find this quite jarring - as at this stage I haven't even started typing yet so don't expect to be reprimanded just yet!

<img width="444" alt="image" src="https://github.com/user-attachments/assets/f51e59f3-bd60-4d9d-9bda-a4a8e4e78d94">

### Steps to reproduce

Have a form with required fields and focus into the first one.

### Form settings

- Multi-page form: No
- Submission Method: Ajax
- Client-side Validation: Yes
- Custom Form Templates: Not for this particular field

### Craft CMS version

5.4.3

### Plugin version

3.0.6

### Multi-site?

No

**Comment by engram-design** [comment]
It shouldn't on focus, but on blur (when focus is lost), but only when the "Validate When Typing" form setting is enabled. Otherwise, client-side validation should only happen when submitting the form.

If the form is submitted and there are errors, the validation becomes "live" however. This is for good UX, which for example on-submit you receive an error that a field is required, and you then fill in that field, you would want the error message removes, rather than hang around until next submit.

Can you clarify what you mean about focus?

**Comment by thisisjamessmith** [comment]
It's happening immediately on focus for me, not on blur. I've tried turning off the "Validate when typing" option, but then validation doesn't happen on blur or focus, only on submit (as you'd expect).

**Comment by engram-design** [comment]
Just to be clear I'm following what you're seeing, here's a video on my end, working as I'd expect it to on blur, only with the validate when typing enabled. 

https://github.com/user-attachments/assets/1633a667-9dc7-4492-acc0-5f149baedcbc

**Comment by thisisjamessmith** [comment]
Thanks, for some reason it's just me then! I'm using custom templates, but only overriding the submit button and checkbox/radio fields so I'm not sure it could be caused by that...? Is there anything in the theme config such as required classes that might be causing it? My themeConfig is using `resetClasses: true` and `formWrapper: false`. I'm also not including any of Formie's CSS files, though I am including the JS.

Edit to add: ah ok, it works fine if I remove my themeConfig.

This is my themeConfig object:

```
{
    resetClasses: true,
    formWrapper: false,
    form: {
        attributes: {
            class: 'js-formieForm fui-form default-links'
        }
    },
    row: {
        attributes: {
            class: 'fui-row mt-4'
        }
    },
    subFieldRow: {
        attributes: {
            class: 'fui-row mt-4'
        }
    },
    nestedFieldRow: {
        attributes: {
            class: 'fui-row mt-4'
        }
    },
    fieldInstructions: {
        attributes: {
            class: '[&>p]:text-sm mt-2'
        }
    },
    fieldRequired: {
        attributes: {
            class: block.sectionBG.value in ['dark'] ? 'text-redOnDarkWCAG' : 'text-red-500'
        }
    },
    fieldError: {
        attributes: {
            class: 'text-red-500 text-xs uppercase font-medium mt-1 rounded bg-red-50 text-sm text-red-700 py-1 px-2' ~ (block.sectionBG.value not in ['dark', 'brand'] ? ' border border-red-200'),
        },
    },
    alertError: {
        attributes: {
            class: 'rounded-md bg-red-50 border border-red-200 text-sm text-red-700 p-4 my-4',
        },
    },
    alertSuccess: {
        attributes: {
            class: 'rounded-md bg-green-50 border-2 border-accent text-sm text-accent-dark p-4 my-4',
        },
    },

    fieldContainer: false,

    agree: {
        fieldInputWrapper: {
            attributes: {
                class: 'mt-2'
            }
        },
        fieldOptionLabel: {
            attributes: {
                class: 'mt-0 group flex'
            }
        },
    },

    checkboxes: {
        fieldOptions: {
            attributes: {
                class: ["{{ field.layout == 'horizontal' ? 'flex gap-4' : 'flex gap-2 flex-col' }}"]
            }
        },
        fieldOptionLabel: {
            attributes: {
                class: 'mt-0 group flex text-sm items-center',
            },
        },
        fieldLabel: {
            resetClass: true,
            attributes: {
                class: 'faux-label',
            },
        },
        fieldInputWrapper: {
            tag: 'fieldset',
            attributes: {
                class: 'mt-4'
            }
        }
    },

    radioButtons: {
        fieldOptions: {
            attributes: {
                class: ["{{ field.layout == 'horizontal' ? 'flex gap-4' : 'flex gap-2 flex-col' }}"]
            }
        },
        fieldOptionLabel: {
            attributes: {
                class: 'mt-0 group flex text-sm items-center',
            },
        },
        fieldLabel: {
            resetClass: true,
            attributes: {
                class: 'faux-label',
            },
        },
        fieldInputWrapper: {
            tag: 'fieldset',
            attributes: {
                class: 'mt-4'
            }
        }
    },

    buttonWrapper: {
        attributes: {
            class: [
                'mt-8 flex justify-start',
                "{{ buttonsPosition == 'right' ? 'justify-end' }}",
                "{{ buttonsPosition == 'center' ? 'justify-center' }}",
                "{{ buttonsPosition == 'left-right' ? 'justify-between' }}",
                "{{ buttonsPosition == 'right-save-left' ? 'justify-start flex-row-reverse' }}",
                "{{ buttonsPosition == 'center-save-left' ? 'justify-center flex-row-reverse' }}",
                "{{ buttonsPosition == 'center-save-right' ? 'justify-center' }}",
                "{{ buttonsPosition == 'save-right' ? 'justify-between' }}",
                "{{ buttonsPosition == 'save-left' ? 'justify-between flex-row-reverse' }}",
            ],
        }
    }
}
```

**Comment by engram-design** [comment]
Interesting to know about your Theme Config. If you're able to send through a reduced config where it's not working, I can take a look!

**Comment by thisisjamessmith** [comment]
Narrowed it down to the `resetClasses: true` line. If I remove that it works as expected. So I'm guessing the validation is hanging off one or more of those `fui-*` classes that Formie forms are peppered with. (I tend to prefix any js-dependent classes with `js-` to avoid this kind of confusion).

**Comment by engram-design** [comment]
Ah right, I see the issue here, it's the lack of `fui-error` class, which we use in our validator, purely to check if an error has already been rendered for the field, and only for live validation.

Fixed for the next release. To get this early, run `composer require verbb/formie:"dev-craft-5 as 3.0.6"`.

**Comment by thisisjamessmith** [comment]
Ah amazing thanks Josh.

**Comment by engram-design** [comment]
Fixed in [3.0.7](https://github.com/verbb/formie/releases/tag/3.0.7)
